spec/helpers/request_methods_spec.rb

```diff
@@ -0,0 +1,188 @@
+      endpoint = 'tests'
+      TestRequester.expects(:entity_name).returns(endpoint)
+      expect(TestRequester.endpoint).to eq(endpoint)
+    end
+  end
+
+  describe 'client' do
+    context 'without a defined address' do
+      it 'raises a NotImplementedError' do
+        expect { TestRequester.client }.to raise_error(NotImplementedError)
+      end
+    end
+
+    context 'with a defined address' do
+      let(:address) { 'http://localhost:3000' }
+
+      before do
+        TestRequester.expects(:address).returns(address)
+      end
+
+      it 'returns a Faraday::Connection' do
+        expect(TestRequester.client).to be_a(Faraday::Connection)
+      end
+    end
+  end
+
+  describe 'request' do
+    context 'with successful responses' do
+      let(:exists_response) { { 'exists' => false } }
+      let(:entities_response) { { 'entities' => { 'id' => 1 } } }
+      let(:prefix_entities_response) { { 'entities' => { 'id' => 2 } } }
+      let(:faraday_stubs) { Faraday::Adapter::Test::Stubs.new }
+      let(:connection) { Faraday.new { |builder| builder.adapter :test, faraday_stubs } }
+
+      before :each do
+        TestRequester.expects(:client).at_least_once.returns(connection)
+        TestRequester.expects(:endpoint).at_least_once.returns('entities')
+      end
+
+      after :each do
+        faraday_stubs.verify_stubbed_calls
+      end
+
+      context 'without an id parameter' do
+        context 'without a prefix' do
+          it 'is expected to make the request without the prefix' do
+            # stub.get receives arguments: path, headers, block
+            # The block should be a Array [status, headers, body]
+            faraday_stubs.get('/entities/') { [200, {}, entities_response] }
+            response = TestRequester.request('', {}, :get)
+            expect(response).to eq(entities_response)
+          end
+        end
+
+        context 'with a prefix' do
+          it 'is expected to make the request with the prefix' do
+            # stub.get receives arguments: path, headers, block
+            # The block should be a Array [status, headers, body]
+            faraday_stubs.get('/prefix/entities/') { [200, {}, prefix_entities_response] }
+            response = TestRequester.request('', {}, :get, 'prefix')
+            expect(response).to eq(prefix_entities_response)
+          end
+        end
+      end
+
+      context 'with an id parameter' do
+        it 'is expected to make the request with the id included' do
+          # stub.get receives arguments: path, headers, block
+          # The block should be a Array [status, headers, body]
+          faraday_stubs.get('/entities/1/exists') { [200, {}, exists_response] }
+          response = TestRequester.request(':id/exists', { id: 1 }, :get)
+          expect(response).to eq(exists_response)
+        end
+      end
+    end
+
+    context 'when the record was not found' do
+      context 'and or the status is 404' do
```

Comment:
This, to me, was really confusing. the test makes an assumption that the request would result in a record not found, but maybe the returning status won't be 404. Or, even, the returning status is 404 but it's not really a record not found.
Shouldn't it be better to standardize this? I would suggest to always treat a 404 as a Record not Found, which would merge these two contexts.